src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -106,16 +619,35 @@ export default function QuestionnaireEditor({ id }: QuestionnaireEditorProps) {
     enabled: !!id,
   });
 
-  const { data: availableOrganizations, isLoading: isLoadingOrganizations } =
-    useQuery({
-      queryKey: ["organizations", orgSearchQuery],
-      queryFn: query(organizationApi.list, {
-        queryParams: {
-          org_type: "role",
-          name: orgSearchQuery || undefined,
-        },
-      }),
-    });
+  const { data: organizations } = useQuery({
+    queryKey: ["questionnaire", id, "organizations"],
+    queryFn: query(questionnaireApi.getOrganizations, {
+      pathParams: { id: id! },
+    }),
+    enabled: !!id,
+  });
+
+  const {
+    data: availableOrganizations,
+    isLoading: isLoadingAvailableOrganizations,
+  } = useQuery({
+    queryKey: ["organizations", orgSearchQuery],
+    queryFn: query(organizationApi.list, {
+      queryParams: {
+        org_type: "role",
+        name: orgSearchQuery || undefined,
+      },
+    }),
+  });
+
+  const { data: availableTags, isLoading: isLoadingAvailableTags } = useQuery({
+    queryKey: ["tags", tagSearchQuery],
```

Comment:
let's keep the key more verbose. this could be mixed up with other tags (maybe patient tags when it comes up)

```suggestion
    queryKey: ["questinnaireTags", tagSearchQuery],
```

Reply:
https://github.com/ohcnetwork/care_fe/blob/develop/src/components/Questionnaire/ManageQuestionnaireTagsSheet.tsx#L66

Reply:
Alrighty